react/features/app/actions.js

```diff
@@ -31,176 +31,72 @@ declare var APP: Object;
-    // XXX As the mobile/React Native app does not employ config on the
-    // WelcomePage, do not download config.js from the deployment when
-    // navigating to the WelcomePage - the perceived/visible navigation will be
-    // faster.
-    if (!room && typeof APP === 'undefined') {
-        return Promise.resolve();
-    }
+        let protocol = location.protocol.toLowerCase();
+        const { contextRoot, host, room } = location;
 
-    /* eslint-disable no-param-reassign */
+        // The React Native app supports an app-specific scheme which is sure to not
+        // be supported by fetch (or whatever loadConfig utilizes).
+        protocol !== 'http:' && protocol !== 'https:' && (protocol = 'https:');
 
-    protocol = protocol.toLowerCase();
+        // TDOO userinfo
 
-    // The React Native app supports an app-specific scheme which is sure to not
-    // be supported by fetch (or whatever loadConfig utilizes).
-    protocol !== 'http:' && protocol !== 'https:' && (protocol = 'https:');
+        const baseURL = `${protocol}//${host}${contextRoot || '/'}`;
+        let url = `${baseURL}config.js`;
 
-    // TDOO userinfo
+        // XXX In order to support multiple shards, tell the room to the deployment.
+        room && (url += `?room=${room.toLowerCase()}`);
 
-    const baseURL = `${protocol}//${host}${contextRoot || '/'}`;
-    let url = `${baseURL}config.js`;
+        let config;
 
-    // XXX In order to support multiple shards, tell the room to the deployment.
-    room && (url += `?room=${room.toLowerCase()}`);
-
-    /* eslint-enable no-param-reassign */
+        try {
+            config = await loadConfig(url);
+        } catch (error) {
+            config = restoreConfig(baseURL);
+            dispatch(loadConfigError(error, locationURL));
+        }
 
-    return loadConfig(url).then(
-        /* onFulfilled */ config => {
-            // FIXME If the config is no longer needed (in the terms of
-            // _loadConfig) and that happened because of an intervening
-            // _loadConfig for the same baseURL, then the unneeded config may be
-            // stored after the needed config. Anyway.
+        if (config) {
             dispatch(storeConfig(baseURL, config));
 
-            return config;
-        },
-        /* onRejected */ error => {
-            // XXX The (down)loading of config failed. Try to use the last
-            // successfully fetched for that deployment. It may not match the
-            // shard.
-            const config = restoreConfig(baseURL);
+            if (getState()['features/base/config'].locationURL === locationURL) {
+                dispatch(setLocationURL(locationURL));
+                dispatch(setConfig(config));
 
-            if (config) {
-                return config;
+                dispatch(setRoom(room));
             }
 
-            throw error;
-        });
+            dispatch(loadConfigError(new Error('Config no longer needed!'), locationURL));
```

Comment:
Should the dispatching of loadConfigError be guarded behind some kind of check? Seems like this might always fire if config is defined.

Reply:
Or maybe it should be removed because you already dispatch the error above?

Reply:
No, it's a bug... I used to have return statements there, so it wasn't invoked after the ```if``` but now they are gone. Fixed it, thanks!